### ISSUE TITLE
login sudo openssh-8.4 version breaks

### ISSUE THREAD

**Issue by brcisna** [ltsp/ltsp#358]
Hello All,

Debian Bullseye
Pretty sure have confirmed that openssh-8.4 update (as of about a month ago) breaks pamltsp
Findings: Done a fresh install on a second server box, with no other added programs,etc., of  Debian Bullseye,(which have used for about one year now)
After installation if i log into client machine with either of two users,,have been added to sudo, sudo command does not work "Login Incorrect". Also if either of the users does a simple logout ,then tries to log back into desktop, "Login Incoorect" in the greeter window.
A reboot of client and user can do a login to desktop with no problems.
One other quirk is, if i purposley enter an incorrect password,for either user one time,then enter correct password, the password is never accepted,and client machine has to be rebooted to get a login to desktop . 
On the ltsp server everything works as expected.
If i delete openssh-8.4 and manually install openssh-8.3 ( a pain) everything works correctly on client. 

Also have read in ubuntu forums of similar anomilies of dodgey logins failing after a period of time, as of openssh-8.4 
Machine is still plenty usable just need to know of this.

Thank You

**Comment by alkisg** [comment]
@brcisna, you're correct, I verified this in my bullseye VM.
Additionally, I verified that installing the previous version works around the problem.

1) Add this to /etc/apt/sources.list:
`deb http://snapshot.debian.org/archive/debian/20201001T000000Z/ `
2) Run:
`apt -o Acquire::Check-Valid-Until=false -o Acquire::Check-Date=false update`
`apt install openssh-client=1:8.3p1-1 openssh-server=1:8.3p1-1 openssh-sftp-server=1:8.3p1-1`

From a first look, it appears that the 8.4 ssh client doesn't work correctly when /home/username is already mounted over ssh. Thanks, I'll troubleshoot this more.

**Comment by alkisg** [comment]
@brcisna, can you test if running these commands on your server, and rebooting the clients, solves the problem?

```shell
sudo sed 's/DISPLAY=$/&" "/' -i /usr/share/ltsp/client/login/pamltsp
sudo ltsp initrd
```

The old ssh was calling our ssh-askpass program when DISPLAY was set to any value, even an empty value.
The new ssh needs a value there, so the sed command above changes `DISPLAY=` to `DISPLAY=" "` in our pamltsp script.

**Comment by brcisna** [comment]
Ok let me try these two commands and ill report back,,,

Barry

**Comment by brcisna** [comment]
Hi alkisg,

Happy to report the above two commands, fixing up the pamltsp, resolved all problems.



1. client sudo commans works as expected
2. client user logout ,then login again works as expected 
3. client using wrong password,then correct password at login greeter now works as expected

So looks like the pamltsp fixup is easier solution versus downgrading openssh#

Thank You so much!

**Comment by alkisg** [comment]
Great, thanks for the report and the feedback, I'll commit this over the weekend.

**Comment by martadinata666** [comment]
Also add confirmed fix. Thanks.

**Comment by alkisg** [comment]
Fix committed in adaed71.
